### PR TITLE
Bugfix - Trigger configuration does not list repositories

### PR DESCRIPTION
**Pull request by talarian1**
Fixes [TCAP-1165](https://www.jfrog.com/jira/browse/TCAP-1165).

### CODE REVIEW

File: ci/pipelines.release.yml [hunk 1]
```diff
@@ -35,48 +36,53 @@ pipelines:
             - export CI=true
             - export JFROG_CLI_BUILD_NAME=teamcity-artifactory-plugin
             - export JFROG_CLI_BUILD_NUMBER=$run_number
-            # Set M2_HOME
-            - MVN_PATH=`which mvn` && export M2_HOME=`readlink -f $MVN_PATH | xargs dirname | xargs dirname`
+            - export JFROG_CLI_BUILD_PROJECT=ecosys
+
```
Comment (yahavi) [inline ci/pipelines.release.yml:40]:
```suggestion
```

File: ci/pipelines.release.yml [hunk 2]
```diff
@@ -35,48 +36,53 @@ pipelines:
...
-              --server-id-deploy ojo --repo-deploy-releases oss-release-local --repo-deploy-snapshots oss-snapshot-local
+            - curl -fL https://install-cli.jfrog.io | sh
+            - jf c rm --quiet
+            - jf c add internal --url=$int_entplus_deployer_url --user=$int_entplus_deployer_user --password=$int_entplus_deployer_apikey
+            - jf mvnc
+              --repo-resolve-releases ecosys-maven-remote --repo-resolve-snapshots ecosys-maven-remote
+              --repo-deploy-releases ecosys-oss-release-local --repo-deploy-snapshots  ecosys-oss-snapshot-local
+
+            # Run audit
+            - jf audit --fail=false
```
Comment (yahavi) [inline ci/pipelines.release.yml:60]: Let's move audit to below git tag

File: ci/pipelines.release.yml [hunk 3]
```diff
@@ -35,48 +36,53 @@ pipelines:
...
-            - env -i PATH=$PATH M2_HOME=$M2_HOME HOME=$HOME mvn versions:set -DnewVersion="${NEXT_VERSION}" -B
+            - env -i PATH=$PATH mvn versions:set -DnewVersion="${NEXT_VERSION}" -B
             - git commit -am "[artifactory-release] Release version ${NEXT_VERSION} [skipRun]" --allow-empty
             - git tag ${NEXT_VERSION}
-            - git push
-            - git push --tags
 
             # Run install and publish
             - >
               env -i PATH=$PATH M2_HOME=$M2_HOME HOME=$HOME
```
Comment (yahavi) [inline ci/pipelines.release.yml:72]:
```suggestion
              env -i PATH=$PATH HOME=$HOME
```

File: ci/pipelines.release.yml [hunk 4]
```diff
@@ -87,6 +93,7 @@ pipelines:
             - git checkout master
             - git merge origin/release
             - git push
+            - git push --tags
 
           onComplete:
             # Show tests in the *Tests* tab
```
Comment (yahavi) [inline ci/pipelines.release.yml:99]: res_teamcityGit_resourcePath -> res_teamcityReleaseGit_resourcePath

File: ci/pipelines.release.yml [hunk 5]
```diff
@@ -35,48 +36,53 @@ pipelines:
...
-            - ./jfrog rt c ojo --url $int_ojo_rt_url --access-token=$int_ojo_rt_token
-            - ./jfrog rt mvnc
-              --server-id-resolve ojo --repo-resolve-releases jfrog-dependencies --repo-resolve-snapshots jfrog-dependencies
-              --server-id-deploy ojo --repo-deploy-releases oss-release-local --repo-deploy-snapshots oss-snapshot-local
+            - curl -fL https://install-cli.jfrog.io | sh
+            - jf c rm --quiet
+            - jf c add internal --url=$int_entplus_deployer_url --user=$int_entplus_deployer_user --password=$int_entplus_deployer_apikey
+            - jf mvnc
+              --repo-resolve-releases ecosys-maven-remote --repo-resolve-snapshots ecosys-maven-remote
+              --repo-deploy-releases ecosys-oss-release-local --repo-deploy-snapshots  ecosys-oss-snapshot-local
```
Comment (yahavi) [inline ci/pipelines.release.yml:57]:
```suggestion
              --repo-deploy-releases ecosys-oss-release-local --repo-deploy-snapshots ecosys-oss-snapshot-local
```

File: ci/pipelines.snapshot.yml [hunk 6]
```diff
@@ -16,45 +16,54 @@ pipelines:
...
 
             # Configure JFrog CLI
-            - ./jfrog rt c ojo --url $int_ojo_rt_url --access-token=$int_ojo_rt_token
-            - ./jfrog rt mvnc
-              --server-id-resolve ojo --repo-resolve-releases libs-release --repo-resolve-snapshots libs-snapshot
-              --server-id-deploy ojo --repo-deploy-releases oss-release-local --repo-deploy-snapshots oss-snapshot-local
+            - curl -fL https://install-cli.jfrog.io | sh
+            - jf c rm --quiet
+            - jf c add internal --url=$int_entplus_deployer_url --user=$int_entplus_deployer_user --password=$int_entplus_deployer_apikey
+            - jf  mvnc
```
Comment (yahavi) [inline ci/pipelines.snapshot.yml:39]:
```suggestion
            - jf mvnc
```

File: ci/pipelines.snapshot.yml [hunk 7]
```diff
@@ -16,45 +16,54 @@ pipelines:
...
+              --repo-resolve-releases ecosys-maven-remote --repo-resolve-snapshots ecosys-maven-remote
+              --repo-deploy-releases ecosys-oss-release-local --repo-deploy-snapshots  ecosys-oss-snapshot-local
```
Comment (yahavi) [inline ci/pipelines.snapshot.yml:41]:
```suggestion
              --repo-deploy-releases ecosys-oss-release-local --repo-deploy-snapshots ecosys-oss-snapshot-local
```

File: ci/pipelines.snapshot.yml [hunk 8]
```diff
@@ -16,45 +16,54 @@ pipelines:
...
+            # Run audit
+            - jf audit --fail=false
+
+            # Delete former snapshots to make sure the release bundle will not contain the same artifacts
+            - jf rt del "ecosys-oss-snapshot-local/org/jfrog/teamcity/teamcity-artifactory-plugin/*" --quiet
 
             # Run install and publish
             - >
-              env -i PATH=$PATH M2_HOME=$M2_HOME HOME=$HOME
+              env -i PATH=$PATH
```
Comment (yahavi) [inline ci/pipelines.snapshot.yml:51]:
```suggestion
              env -i PATH=$PATH HOME=$HOME
```

File: server/src/main/java/org/jfrog/teamcity/server/global/DeployableArtifactoryServers.java [hunk 9]
```diff
@@ -98,7 +100,13 @@ public List<String> getServerLocalAndCacheRepos(long serverUrlId, boolean overri
                 try {
                     CredentialsBean deployingCredentials = CredentialsHelper.getPreferredDeployingCredentials(serverConfig, overrideDeployerCredentials, username, password);
                     ArtifactoryBuildInfoClient client = getArtifactoryBuildInfoClient(deployingCredentials, serverConfig);
-                    return client.getLocalAndCacheRepositoriesKeys();
+                    List<String> localRepos = client.getLocalRepositoriesKeys();
```
Comment (yahavi) [inline server/src/main/java/org/jfrog/teamcity/server/global/DeployableArtifactoryServers.java:103]: Let's close the clients in all file using the try-with-resources algorithm

**Review by yahavi — CHANGES_REQUESTED** [review]

File: agent/src/main/java/org/jfrog/teamcity/agent/MavenBuildInfoExtractor.java [hunk 10]
```diff
@@ -306,7 +306,7 @@ private void addDependencies(ModuleBuilder moduleBuilder, Element dependenciesEl
                 for (Element idElement : idList) {
                     String idText = idElement.getText();
                     if (StringUtils.isNotBlank(idText)) {
-                        dependencyBuilder.addRequiredBy(idText);
+                        dependencyBuilder.addRequestedBy(new String[]{idText});
```
Comment (yahavi) [inline agent/src/main/java/org/jfrog/teamcity/agent/MavenBuildInfoExtractor.java:309]:
```suggestion
                        dependencyBuilder.addRequestedBy(new String[]{idText, moduleId});
```

**Review by yahavi — APPROVED** [review]
LGTM

**Comment by kasperpagh** [comment]
Thank you for fixing this :-) 

Best regards 
- Someone who wants to use this plugin ;-)

**Review by eyalbe4 — APPROVED** [review]

**Pull request merged**